packages/myst-cli/src/session/session.ts

```diff
@@ -159,13 +160,8 @@ export class Session implements ISession {
 
   plugins: ValidatedMystPlugin | undefined;
 
-  _pluginPromise: Promise<ValidatedMystPlugin> | undefined;
-
-  async loadPlugins() {
-    // Early return if a promise has already been initiated
-    if (this._pluginPromise) return this._pluginPromise;
-    this._pluginPromise = loadPlugins(this);
-    this.plugins = await this._pluginPromise;
+  async loadPlugins(plugins: PluginInfo[]) {
+    this.plugins = await loadPlugins(this, plugins);
```

Comment:
We are no longer caching the `_pluginPromise` for early return. Previously this was required since `loadPlugins` was called during `loadFile` and `loadFile` was called multiple times in parallel.

Since we've moved this to project frontmatter loading, it should only be called once.

(Also, we can't really cache the promise any more anyway, since `loadPlugins` could theoretically be called with different arrays of input `plugins`.)